### PR TITLE
Feat: Implement custom primary color

### PR DESCRIPTION
**Pull request by Biowulf21**
Hi @sadespresso, this is the feature implementation for #105 .

This is my first time submitting a PR for a FOSS repo, so I do apologize in advance if my code is not up to FOSS standard. I'd like to work on this project on my spare time, as I see a lot of potential in this application :smile: .

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -3,20 +3,22 @@ description: A personal finance managing app
...
 
-version: "0.3.3+35"
+version: "0.3.4+0"
 
 environment:
   sdk: ">=3.1.3 <4.0.0"
 
 dependencies:
   auto_size_text: ^3.0.0
+  colornames: ^0.1.0
```
Comment (sadespresso) [inline pubspec.yaml:13]: It seems like `colornames` only supports English...

**Comment by sadespresso** [comment]
Hello @Biowulf21, thank you for taking your time to implement this feature.

It would be more appreciated if we discuss before jumping to implementations. As this is a very young project, I have so many things in mind, and trying my best to list them down in the issues.

Moving forward with your PR, we have few problems.

* First of all, `colornames` package seem to support only English, but Flow plans to support multiple languages. So it's best if we define the names locally, or find a library that supports multiple languages. Not having names is fine too.
* As for the color picker, I was thinking to handpick number of colors and offer them, and they would have different perceived brightness in light/dark mode. This makes localization more easy, and allows us to create different app icons for each color in iOS.

**Comment by sadespresso** [comment]
This, for example..

image

File: assets/l10n/en_US.json [hunk 2]
```diff
@@ -18,6 +18,7 @@
     "general.delete.all": "Delete all",
     "general.copy.success": "Copied to clipboard",
     "general.copy.clickToCopy": "Click to copy",
+    "general.confirm.okay": "Okay",
```
Comment (sadespresso) [inline assets/l10n/en_US.json:21]: I don't think this is necessary, use `done` or `confirm` instead

**Review by sadespresso — CHANGES_REQUESTED** [review]
I will have this in Flow if you make following changes:

* Remove `colornames` package, and not use color names for now or use a localized color name provider
* (MAYBE) not require restart after changing theme color. You can write sth in `Flow` (the main app widget) and have it transcend the changes. Nice to have this, but not necessary.
* Provide a video demonstrating the implemented feature

**Comment by sadespresso** [comment]
Again, thank you for the PR!

Let me know if you want to proceed with the suggestions and review

**Comment by Biowulf21** [comment]
Hello, @sadespresso. I'll implement these changes as soon as I get home. I'm out of the house at the moment.

As for communicating features moving forward, I'd like to connect with you on discord if that's alright with you.

Thanks.

**Comment by sadespresso** [comment]
:D :D 

We have a [discord channel](https://discord.gg/Ndh9VDeZa4), and you can find me, or my account in there!

File: pubspec.yaml (hunk 1 above)
Comment (Biowulf21) [reply pubspec.yaml]: Hello @sadespresso, sorry for the delay. I've been busy with life lately. Anyway, I've removed the `colornames` package as a dependency with my latest changes

File: assets/l10n/en_US.json (hunk 2 above)
Comment (Biowulf21) [reply assets/l10n/en_US.json]: I've removed this localization message as well as the entire dialog. I followed your advice on using `Flow.of(context)` instead so the dialog is no longer needed.

**Comment by Biowulf21** [comment]
Hello @sadespresso  can you point me into the right direction as to which color picker we'll be using? I can refactor my current implementation if you already have a package in mind.

Thanks.

**Comment by sadespresso** [comment]
Hello @Biowulf21 

Thank you so much for the work! I really appreciate it.

I will check it out once I have time, should be soon.

**Review by sadespresso — APPROVED** [review]

**Comment by sadespresso** [comment]
LGTM.

Thank you!

-----------

In the future, I plan to give option to use `dynamic colors`. It's the new Android 13 thing I believe...

Also, in the future, we'd have much more simpler color selector.

**Pull request merged**